src/transformers/modeling_utils.py

```diff
@@ -1299,18 +1346,7 @@ def _generate_no_beam_search(
                     [attention_mask, attention_mask.new_ones((attention_mask.shape[0], 1))], dim=-1
                 )
 
-        # if there are different sentences lengths in the batch, some batches have to be padded
-        if sent_lengths.min().item() != sent_lengths.max().item():
```

Comment:
I think we need this for specific cases - do you think we can get rid of it? @yjernite

Reply:
we already pad above and the tests seem to pass. The bart slow test has an unevenly lengthed batch, so I think we're OK.

Reply:
@sshleifer Note that this is part of `_generate_no_beam_search`, so it is not tested in Bart. Also, it doesn't matter if the `input_ids` for the encoder have an unevenly lengthed batch since generation only concerns with the decoder.

I looked a bit more into it and thought of a bunch of cases and I agree that I don't think this is necessary. I remember vaguely though that I kept it for some reason to capture a special case, but maybe I was just wrong before. 

So good for me to remove this for now.

Also it's important to note that the tests are randomized (since they use sampling, so it might very well be that the tests are fine in one run, but not anymore in the other one). So in case we have flaky generation tests, this could be a reason. Also linking the PR that added this chunk of code for reference: https://github.com/huggingface/transformers/pull/2885.
It also seems to me that we can remove it